### DISCUSSION TITLE
我正在做一个二次封装的小项目, 为了避免自定义元素过大, 引入不必要的部分, 除了写eslint, 还有什么建议?

### DISCUSSION THREAD

**Discussion by yoyo837** [hellof2e/quark-design#236] [Ideas]
![image](https://user-images.githubusercontent.com/6134547/227870102-f42c0b64-3265-497a-ab95-0e9f8e877c1c.png)

**Comment by sanqi-med** [comment]
采用按需引入即可，就是你现在的这种方式

**Reply by yoyo837** [reply]
我的意思是开发者可能还不小心、或者不规范、又或者是之前有使用后面逻辑调整未删除的引入, 如:

```js
import 'quarkd/lib/button';
import 'quarkd/lib/xxx';
```

实际模版只使用了 `button`, 未使用 `xxx`, 的场景, 是否只能依赖写 eslint 插件?

**Reply by sanqi-med** [reply]
这个因为使用方已经自己 import 了，组件内部是感知不到是否有使用， eslint 也不好处理这个问题，因为 web components 只要求自定义元素被注册就可以了，而不像 import  xxx from 'quarkd/lib/xxx'， 这种是可以通过eslint 来处理

**Reply by yoyo837** [reply]
那看来确实只能自己项目写eslint插件来检查不必要的`import`了, 3Q.
